src/Build/BackEnd/BuildManager/BuildManager.cs

```diff
@@ -2602,27 +2602,22 @@ private void PerformSchedulingActions(IEnumerable<ScheduleResponse> responses)
                         break;
 
                     case ScheduleActionType.CreateNode:
-                        var newNodes = new List<NodeInfo>();
+                        IList<NodeInfo> newNodes = _nodeManager.CreateNodes(GetNodeConfiguration(), response.RequiredNodeType, response.NumberOfNodesToCreate);
 
-                        for (int i = 0; i < response.NumberOfNodesToCreate; i++)
+                        if (newNodes?.Count != response.NumberOfNodesToCreate || newNodes.Any(n => n == null))
```

Comment:
Could there be a case where multiple requester ask for max nodes, but only a subset would spawn?

Reply:
In previous code, when it was called in for loop, if any of loop iteration had failed to acquire node, whole build have had failed on `BuildAbortedException`. From this I have drawn logical conclusion that if not all nodes has been crated new code is supposed to failed same way.